src/v/pandaproxy/api/api-doc/schema_registry.json

```diff
@@ -181,6 +181,82 @@
         }
       }
     },
+"/config/{subject}/iceberg": {
```

Comment:
Is this the first non standard extension we've added? I would vote for putting our extensions under a redpanda namespace to prevent any clashes with other implementations

Reply:
> the first...?

great question. i think so but not 100% sure

> putting our extensions under a redpanda namespace

great idea either way

Reply:
Waffled on this a bit. are you imagining a scenario where _everything_ in SR would also be used by iceberg? or more of a guardrail so users don't forget to turn it on per subject?

Incidentally, it should be fairly easy to extend the proposed implementation, but I'm still not sure how agreeable this "piggybacking" approach is to others. e.g. @BenPope had suggested shoving this bit of state into the `metadata` field of the schema itself.

Reply:
Yeah a few customers are planning on only using SR for iceberg, so I think it's worthwhile.

> suggested shoving this bit of state into the metadata field of the schema

I don't really know the storage mechanism here, but this is subject level metadata - not schema level?

Reply:
Ok cool, understood.

> this is subject level metadata

Yeah, it is. The more i think about it the more I feel something close to this PR (+ a global mode) is basically sufficient, but I'm not wedded to it.

 There's been some motion in SR lately, so I'm not pushing on this PR atm. Look forward to hearing feedback from Ben & whomever else.